vyper/old_codegen/parser_utils.py

```diff
@@ -526,27 +599,56 @@ def make_setter(left, right, pos):
     elif isinstance(left.typ, ByteArrayLike):
         # TODO rethink/streamline the clamp_basetype logic
         if _needs_clamp(right.typ, right.encoding):
-            _val = LLLnode("val", location=right.location, typ=right.typ)
-            copier = make_byte_array_copier(left, _val, pos)
-            ret = ["with", _val, right, ["seq", clamp_bytestring(_val), copier]]
+            with right.cache_when_complex("bs_ptr") as (b, right):
+                copier = make_byte_array_copier(left, right, pos)
+                ret = b.resolve(["seq", clamp_bytestring(right), copier])
         else:
             ret = make_byte_array_copier(left, right, pos)
 
         return LLLnode.from_list(ret)
 
+    elif isinstance(left.typ, DArrayType):
+        if not _typecheck_list_make_setter(left, right):
+            return
+
+        # handle literals
+        if right.value == "multi":
+            return _complex_make_setter(left, right, context, pos)
+
+        # TODO should we enable this?
+        # implicit conversion from sarray to darray
```

Comment:
implicit conversion for variables? probably not
implicit conversion for constants? maybe, but probably not
implicit conversion for literals? almost definitely necessary

Reply:
literals are handled here:
https://github.com/vyperlang/vyper/blob/b7b5e189b30b7a2fd2aa07a38d6aed43b9689101/vyper/old_codegen/expr.py#L1054-L1057